pylops/signalprocessing/dtcwt.py

```diff
@@ -119,27 +115,33 @@ def _coeff_to_array(self, pyr: dtcwt.Pyramid) -> NDArray:
         return flat_coeffs
 
     def _array_to_coeff(self, X: NDArray) -> dtcwt.Pyramid:
-        lowpass = np.array([x.real for x in X[-1 * self.lowpass_size :]]).reshape(
-            (-1, self.second_dim)
+        lowpass = np.array([x.real for x in X[-self.lowpass_size :]]).reshape(
+            (-1, self.first_dim)
         )
         _ptr = 0
         highpasses = ()
         for _sl in self.slices:
-            _h = X[_ptr : _ptr + (_sl * self.second_dim)]
-            _ptr += _sl * self.second_dim
-            _h = _h.reshape((-1, self.second_dim))
+            _h = X[_ptr : _ptr + (_sl * self.first_dim)]
+            _ptr += _sl * self.first_dim
+            _h = _h.reshape((-1, self.first_dim))
             highpasses += (_h,)
         return dtcwt.Pyramid(lowpass, highpasses)
 
     def get_pyramid(self, X: NDArray) -> dtcwt.Pyramid:
+        """Return Pyramid object from transformed array
+        """
         return self._array_to_coeff(X)
 
     @reshaped
     def _matvec(self, x: NDArray) -> NDArray:
+        x = x.swapaxes(self.axis, -1)
```

Comment:
@this is not good, you should use `swapaxes` like here https://github.com/PyLops/pylops/blob/e4d914abb7e81eca3ad46f1508c99848771bc4bb/pylops/basicoperators/firstderivative.py#L139

When something is readily available and used in many other operators we should not deviate from it unless there is a special requirement, I do not see it here :)